flycheck.el

```diff
@@ -5351,6 +5353,30 @@ See URL `http://phpmd.org/' for more information about phpmd."
                       errors)))))))))
        (nreverse errors)))))
 
+(defun flycheck-parse-tslint (output checker buffer)
+  "Parse TSLint errors from JSON OUTPUT.
+
+CHECKER and BUFFER denoted the CHECKER that returned OUTPUT and
+the BUFFER that was checked respectively.
+
+See URL `https://palantir.github.io/tslint/' for more information
+about TSLint."
+  (let ((json-object-type 'plist))
+    (let ((errors)
+	  (tslint-json-output (json-read-from-string output)))
+      (dotimes (n (length tslint-json-output))
+	(let* ((emessage (aref tslint-json-output n))
+	       (position (plist-get emessage :startPosition)))
+          (push (flycheck-error-new-at
+                 (+ 1 (plist-get position :line))
+                 (+ 1 (plist-get position :character))
+                 'warning (plist-get emessage :failure)
+                 :id (plist-get emessage :ruleName)
+                 :checker checker
+                 :buffer buffer
+                 :filename (plist-get emessage :name)) errors)))
+      (nreverse errors))))
```

Comment:
Is there a reason why you're using plists as the output type here?  I think it'd be easier to read if you used alists in combination with `let-alist`.  Take a look at the existing XML error parsers which make use of `let-alist` for an example.